src/vos/tests/evt_ctl.c

```diff
@@ -1682,6 +1683,239 @@ test_evt_various_data_size_internal(void **state)
 	}
 }
 
+static void
+test_evt_node_size_internal(void **state)
+{
+	struct test_arg		*arg = *state;
+	daos_handle_t		 toh;
+	int			order_size, rc;
+	bool			evt_created;
+
+	for (order_size = 0; order_size < ORDER_TEST_SIZE; order_size++) {
+		evt_created = false;
+		rc = evt_create(arg->ta_root, ts_feats, order_size,
+				arg->ta_uma, &ts_evt_desc_cbs, &toh);
+		if ((order_size >= EVT_MIN_ORDER) &&
+		(order_size <= EVT_MAX_ORDER)) {
+			assert_int_equal(rc, 0);
+			evt_created = true;
+		} else {
+			assert_int_not_equal(rc, 0);
+		}
+		if (evt_created) {
+			rc = evt_destroy(toh);
+			assert_int_equal(rc, 0);
+		}
+	}
+}
+
+/* Args : Input : string_val pointer
+*                 num: number of time to repeat string val
+*         Output: ret_string: Returns concatenated string.
+*/
+void repeat_string(char *ret_string, char *string_val, int num)
```

Comment:
Without buffer sizes this seems pretty unsafe. It's seems to only be used in this test file so probably okay, but callers have to ensure enough space is allocated.

Reply:
Added a note for now. 
*		  NOTE: pre-allocate ret_string buffer
*		  before calling this function. Otherwise,
*		  you will get segmentation fault.